src/client.rs

```diff
@@ -93,101 +67,16 @@ impl DatagoClient {
             return;
         }
 
-        // Spawn a new thread which will query the DB and send the pages
```

Comment:
all of the orchestration code is gone from "client.rs", which can be focused on higher level tasks